backend/src/core/middlewares/file-attachment.middleware.ts

```diff
@@ -156,11 +161,44 @@ async function streamCampaignEmbed(
   return res
 }
 
+async function uploadFileToPresignedUrl(
+  req: Request,
+  res: Response
+): Promise<Response> {
+  // 1. Get uploaded file from request
+  const uploadedFile = req.files?.file as fileUpload.UploadedFile
+  if (!uploadedFile) {
+    return res
+  }
+  // 2. Get presigned URL for file upload
+  const { presignedUrl, signedKey } = await UploadService.getUploadParameters(
+    uploadedFile.mimetype
+  )
+  try {
+    // 3. Upload file to presigned URL
+    const response = await axios.put(presignedUrl, uploadedFile.data, {
+      headers: {
+        'Content-Type': uploadedFile.mimetype,
+      },
+      withCredentials: false,
+    })
+    // 4. Return the etag and transactionId to the FE
+    const formattedEtag = removeFirstAndLastCharacter(response.headers.etag)
```

Comment:
Why do you do this, is it because:

```
Removing the first and last character from an ETag is a common technique used to address caching issues. ETags are used in HTTP headers to help with web caching. By removing the first and last character, your friend might be attempting to modify the ETag in order to force a cache revalidation. This can be useful when there are issues with caching and the content is not being updated as expected.
```